network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -115,7 +122,34 @@ struct ConnectionState<T> {
     required_services: Services,
 }
 
-impl<T: Ord> ConnectionState<T> {
+impl<T: Clone + Ord + Unpin> Future for ConnectionState<T> {
+    type Output = ();
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let mut unban_id = None;
+
+        if let Some((id, sleep)) = self.unban_timeout.as_mut() {
```

Comment:
```suggestion
        while let Some((id, sleep)) = self.unban_timeout.as_mut() {
```

Better make this a `while` loop so that every created sleep in there is actually polled until it returns `Poll::Pending`. Needs to have a `break;` in the `else` of `if sleep.poll_unpin(cx).is_ready()` though.